app/models/grda_warehouse/tasks/client_cleanup.rb

```diff
@@ -471,6 +550,14 @@ def choose_best_gender dest_attr, source_clients
         end
       end
 
+      # Set DifferentIdentityText
+      additional = sorted_source_clients.map { |m| m[:DifferentIdentityText] }.compact
+      dest_attr[:DifferentIdentityText] = if additional.empty?
+        nil
+      else
+        additional.first
+      end
```

Comment:
do we need to handle blank strings (s/compact/compact_blank/)? Also maybe could golf down the assignment conditional
```suggestion
      additional = sorted_source_clients.map { |m| m[:DifferentIdentityText] }.compact_blank
      dest_attr[:DifferentIdentityText] = additional.presence&.first
```